src/components/InputProblem.tsx

```diff
@@ -1,7 +1,92 @@
+import React, { useState, useCallback } from 'react'
 import { BaseProblemProps } from './ProblemSetBlock'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import { mathifyElement } from '../lib/math'
+import * as Yup from 'yup'
 
-interface InputProblemProps extends BaseProblemProps {}
+interface InputProblemProps extends BaseProblemProps {
+  comparator: string
+}
+
+interface InputSchema {
+  response: string | number
+}
+
+interface InputFormValues {
+  response: string
+}
+
+export const InputProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback,
+  solution, retryLimit, content, comparator, encourageResponse, buttonText, correctResponse
+}: InputProblemProps): JSX.Element => {
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const [inputDisabled, setInputDisabled] = useState(false)
+  const [feedback, setFeedback] = useState('')
+
+  const schema = (): Yup.SchemaOf<InputSchema> => {
+    if (comparator === 'integer') {
+      return Yup.object({
+        response: Yup.number().integer().typeError('Please provide an Integer').required('Please provide an Integer')
```

Comment:
I found this randomly, but it seems if you provide an input like `24.5` you get the message `response must be an integer`. I think that's the default for the `integer()` here, so if you wanted to put a custom message you can make that `integer("Please provide an Integer not a float")` or something like that.